utilities/calibration-delta-updater/script/blf-calibration-delta-updater.py

```diff
@@ -35,69 +33,153 @@ def open_polydriver(robot_name, control_board, joints_list):
+    if not sensor_bridge.advance():
+        raise RuntimeError('Unable to read the sensors')
 
-    tree = etree.parse(args.input)
-    root = tree.getroot()
-    offsets = get_offsets(root)
+    _, joints_values, _ = sensor_bridge.get_joint_positions()
 
-    control_board, sensor_bridge = open_polydriver(args.robot, args.board, args.joints)
+    # compute the offset
+    delta = joints_values[joint_index] * 180 / math.pi - expected_values[joint_index]
+
+    tolerance = tolerances[joint_index]
+    if abs(delta) < tolerances[joint_index]:
+        print_info("The offset is equal to " + str(delta)
+                   + " deg. The absolute value is lower than the desired tolerance "
+                   + str(tolerance) + " deg.")
+        print_info("The offset is set to 0 deg")
+        return 0.0
+    else:
+        return delta
+
+
+def parse_handler(file, robot_part):
+    # load the parameter handler
+    parameters = blf.parameters_handler.YarpParametersHandler()
+
+    if os.path.isfile(file):
+        if not parameters.set_from_file_path(file):
+            raise RuntimeError('Unable to load the configuration file')
+    else:
+        if not parameters.set_from_filename(file):
+            raise RuntimeError('Unable to load the configuration file')
+
+    group = parameters.get_group(robot_part)
+    if group is None:
+        raise RuntimeError('Unable to find the group named ' + robot_part)
+
+    joints = group.get_parameter_vector_string('joints_list')
+    board = group.get_parameter_string('control_board')
+    values = group.get_parameter_vector_float('expected_values_in_degrees')
+    tolerance = group.get_parameter_vector_float('tolerance_in_degrees')
+
+    parameters.set_parameter_vector_string('joints_list', joints)
+    parameters.set_parameter_vector_string('remote_control_boards', [board])
+
+    # populate the parameters handler with additional parameters required by the application. There is no need to
+    # expose these parameter in the configuration file
+    parameters.set_parameter_string('local_prefix', 'joint-offset-updater')
+    parameters.set_parameter_bool('check_for_nan', False)
+    parameters.set_parameter_bool('stream_joint_states', True)
+    return parameters, values, joints, tolerance
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Simple tool to semi-automatically update the calibration delta of a YARP-based robot.')
+    parser.add_argument('-i', '--input', type=str, required=True, help='Path to the input xml file containing the calibration deltas')
+    parser.add_argument('-o', '--output', type=str,
+                        required=True, help='Path to the output xml file containing the calibration deltas')
+    parser.add_argument('-p', '--part', type=str, required=True,
+                        help='Name of the group will be loaded in the configuration file. For instance  left_leg or right_leg.')
+    parser.add_argument('--config', type=str, required=False,
```

Comment:
The README documents only the "short" version (e.g. ``-p``). It may be worth noting somewhere that also ``--part`` can be used, for example.